types/types.go

```diff
@@ -107,3 +108,20 @@ func UnpackWarpMessage(unsignedMsgBytes []byte) (*avalancheWarp.UnsignedMessage,
 	}
 	return unsignedMsg, nil
 }
+
+func FetchWarpMessageFromID(ethClient ethclient.Client, warpID common.Hash, blockNum *big.Int) (*WarpMessageInfo, error) {
+	logs, err := ethClient.FilterLogs(context.Background(), interfaces.FilterQuery{
+		Topics:    [][]common.Hash{{WarpPrecompileLogFilter}, nil, {warpID}},
+		Addresses: []common.Address{warp.ContractAddress},
+		FromBlock: blockNum,
+		ToBlock:   blockNum,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch logs: %w", err)
+	}
+	if len(logs) != 1 {
+		return nil, fmt.Errorf("found more than 1 log: %d", len(logs))
+	}
+
+	return NewWarpMessageInfo(logs[0])
+}
```

Comment:
This feels a little odd to be in the `types` package. I think either `utils` or `relayer` would probably be a better fit for it.